.github/workflows/update-and-rebuild.yaml

```diff
@@ -1,60 +1,116 @@
+    runs-on: ubuntu-latest
+    needs: check-new-finschia-tag
+    if: needs.check-new-finschia-tag.outputs.NEW_TAG
+    env:
+      NEW_TAG: ${{ needs.check-new-finschia-tag.outputs.NEW_TAG }}
+      FINSCHIA_SDK_VERSION: ${{ needs.check-new-finschia-tag.outputs.FINSCHIA_SDK_VERSION }}
+      WASMD_VERSION: ${{ needs.check-new-finschia-tag.outputs.WASMD_VERSION }}
+      IBC_GO_VERSION: ${{ needs.check-new-finschia-tag.outputs.IBC_GO_VERSION }}
+      TENDERMINT_VERSION: ${{ needs.check-new-finschia-tag.outputs.TENDERMINT_VERSION }}
+
+    steps:
+      - name: Check out to finschia-wasm
+        uses: actions/checkout@v4
+        with:
+          repository: Finschia/finschia-wasm
+
+      - name: fetch and merge submodule
+        run: |
+          git submodule update --init --remote
+          git submodule status
 
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: rust setup
+        run: rustup component add rustfmt
 
-      - name: Set up cargo cache
-        uses: actions/cache@v3
-        continue-on-error: false
+      - name: buf setup
+        uses: bufbuild/buf-setup-action@v1
+
+      - name: Install protoc
+        uses: arduino/setup-protoc@v1
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-cargo-
-
-      - name: Generate code
+          version: "3.14.0"
+
+      - name: Install buf
         run: |
-          git config --global user.name "github-actions[bot]"
-          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          GO111MODULE=on go install github.com/bufbuild/buf/cmd/buf@v1.15.1
 
-          # buf is installed in GOPATH
-          export PATH=${PATH}:`go env GOPATH`/bin
+      - name: use actions cache
+        uses: ./.github/actions/cache
 
-          ./scripts/update-and-rebuild.sh "$OSMOSIS_REV"
+      - name: set each submodule version to env file in finschia-wasm
+        run: |
+          echo '# Choose from https://github.com/Finschia/finschia/releases' | cat > env
+          echo FINSCHIA_VERSION=${{ env.NEW_TAG }} | cat >> env
+          echo FINSCHIA_SDK_VERSION=${{ env.FINSCHIA_SDK_VERSION }} | cat >> env
+          echo WASMD_VERSION=${{ env.WASMD_VERSION }} | cat >> env
+          echo TENDERMINT_VERSION=${{ env.TENDERMINT_VERSION }} | cat >> env
+          echo IBC_GO_VERSION=${{ env.IBC_GO_VERSION }} | cat >> env
 
-        env:
-          OSMOSIS_REV: ${{ inputs.osmosis_rev }}
+      - name: Generate new packages/finschia-std
+        run: |
+          cargo run
+        working-directory: packages/proto-build
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: Automated bump up ${{ env.NEW_TAG }}
```

Comment:
how about adding a tag (maybe `build: `) for semantic commit messages?

Reply:
Thanks. Fixed here.
[fix: add the tag "build:" to the commit message of the auto bumpup pr](https://github.com/Finschia/finschia-wasm/pull/9/commits/f439cf5eb432cd7197155922f1d8379e9e96797b)